storage/local/persistence.go

```diff
@@ -544,11 +544,15 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 	}
 
 	defer func() {
-		f.Sync()
+		syncErr = f.Sync()
 		closeErr := f.Close()
 		if err != nil {
 			return
 		}
+		err = syncErr
+		if err != nil {
+			return
+		}
```

Comment:
Yes, sorry... I'm just repeating history here. Went through that when I wrote it. :(